tests/test_hive.py

```diff
@@ -0,0 +1,190 @@
+import json
+import logging
+import os
+import pytest
+import tempfile
+import time
+
+import sdk_auth
+import sdk_cmd
+import sdk_marathon
+import sdk_utils
+
+import spark_utils
+
+
+log = logging.getLogger(__name__)
+
+GENERIC_HDFS_USER_PRINCIPAL = "hdfs@{realm}".format(realm=sdk_auth.REALM)
+ALICE_USER = "alice"
+ALICE_PRINCIPAL = "{user}@{realm}".format(user=ALICE_USER, realm=sdk_auth.REALM)
+HIVE_APP_ID = "cdh5-hadoop-hive-kerberos"
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class KerberosSetup:
+    def __init__(self, kerberos_env, hive_agent_ip, hive_agent_hostname):
+        self.kerberos_env = kerberos_env
+        self.hive_agent_ip = hive_agent_ip
+        self.hive_agent_hostname = hive_agent_hostname
+
+
+class HadoopSetup:
+    def __init__(self, kerberos_setup, hdfs_base_url, hive_config_url):
+        self.kerberos_setup = kerberos_setup
+        self.hdfs_base_url = hdfs_base_url
+        self.hive_config_url = hive_config_url
+
+
+def _get_agent_ip():
```

Comment:
It's not quite clear what this function is supposed to do? Get the IP of the first private agent?

What about using this: https://github.com/mesosphere/dcos-commons/blob/master/testing/sdk_agents.py#L32